# Cálculo de Estatísticas de Imagens
Este notebook calcula a média e o desvio padrão das imagens extraídas dos vídeos.

In [1]:
import os
import numpy as np
from PIL import Image
from torchvision.transforms import ToTensor
from torch.utils.data import DataLoader, Dataset

class ImageDataset(Dataset):
    def __init__(self, root_dir, transform=None):
        self.root_dir = root_dir
        self.transform = transform
        self.image_paths = []
        for subdir, _, files in os.walk(root_dir):
            for file in files:
                if file.endswith(('.jpg', '.png', '.jpeg')):
                    self.image_paths.append(os.path.join(subdir, file))

    def __len__(self):
        return len(self.image_paths)

    def __getitem__(self, idx):
        img_path = self.image_paths[idx]
        image = Image.open(img_path).convert('RGB')
        if self.transform:
            image = self.transform(image)
        return image

def calculate_mean_std(root_dir):
    transform = ToTensor()
    dataset = ImageDataset(root_dir, transform=transform)
    loader = DataLoader(dataset, batch_size=64, shuffle=False)
    mean = 0.0
    std = 0.0
    n_samples = 0
    for images in loader:
        batch_samples = images.size(0)
        images = images.view(batch_samples, images.size(1), -1)
        mean += images.mean(2).sum(0)
        std += images.std(2).sum(0)
        n_samples += batch_samples
    mean /= n_samples
    std /= n_samples
    print(f"Mean: {mean}")
    print(f"Std: {std}")

In [2]:
# Exemplo de uso
ROOT_DIR = "frames"
calculate_mean_std(ROOT_DIR)

Mean: tensor([0.4434, 0.4797, 0.5037])
Std: tensor([0.1893, 0.1636, 0.1239])


Resultados (Apos adicionar packtage 02 )

Mean: tensor([0.4434, 0.4797, 0.5037])

Std: tensor([0.1893, 0.1636, 0.1239])